### DISCUSSION TITLE
Feature: DECDHL/DECSWL/DECDWL

### DISCUSSION THREAD

**Discussion by paulie-g** [ghostty-org/ghostty#3799] [Feature Requests, Ideas]
DECDHL in particular is the ideal way to implement headings in text browsers, man page viewers, markdown renderers and so on. It is supported by xterm + descendants, mlterm, Konsole + descendants, and a few others. While it could be written off as 'not commonly used', this is only the case given spotty support (which is changing, cf Konsole's recent implementation). In a world where notcurses exists, useful but less-than-universally supported features will eventually get exercised (note: I can not speak for author of notcurses as to when/if DECDHL support will be added). I understand that this is difficult to implement for emulators who use a packed texture atlas for GPU rendering and have a lot of assumptions about cell size all over the place.

**Comment by pluiedev** [comment]
See also #5563 for a more modern alternative protocol
